lib/new_relic/agent/pipe_channel_manager.rb

```diff
@@ -85,12 +97,41 @@ def deserialize_message_length(data)
         end
 
         def write(data)
+          if nr_debug?
+            lock_granted = @write_lock.try_lock
+
+            NewRelic::Agent.logger.error(
+              "Critical code section violated (Thread ID: #{Thread.current.object_id})\n" +
+              "Backtrace:\n" + caller.join("\n").gsub(/^/, "\t")
+            ) unless lock_granted
+
+            unless @file
+              @file = File.open('/tmp/marshall_log', 'wb')
```

Comment:
Different agents will continually write over this file.  There are multiple pipe channel managers, I believe.  We need to generate either random file names or those unique to the endpoints attached.

Reply:
Nope. In resque we have only one. I tested it on my laptop, seems to works well